### PR TITLE
Filter pokemon by size

### PR DESCRIPTION
**Pull request by Karl-Parris**
Filter pokemon by size

## Description
Displays pokemon size
Adds options for filtering by size

## Motivation and Context
Asked for by local community (to hunt XXS and XXL for badge)

## How Has This Been Tested?
- Tested toggling off "filter by size". Correctly shows all pokemon.
- Tested toggling on "filter by size" and selected multiple size options. Correctly only displays pokemon of those sizes
- Tested with pokemon data that has no size. They are filtered out when you have a size filter on, but show fine when filter is off. 

## Screenshots (if appropriate):
image

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by JimBim89** [comment]
Thanks for this, with showcases being a thing this might be useful.

The size filter settings don't seem to work after reloading the page.
They need to be re-toggled to take effect again, otherwise nice job.

**Comment by na-ji** [comment]
FYI, the size and weight are now random for each player. So it's no longer useful to display them on the map. Niantic did the change probably to prevent the contest from being too easy.

File: static/js/map/map.settings.js [hunk 1]
```diff
@@ -27,6 +27,8 @@ function initSettings() {
     }
     if (serverSettings.pokemonValues) {
         settings.filterPokemonByValues = Store.get('filterPokemonByValues')
+        settings.filterPokemonBySize = Store.get('filterPokemonBySize')
+        settings.filterPokemonBySize = Store.get('filterPokemonBySizeOptions')
```
Comment (pki0) [inline static/js/map/map.settings.js:31]: Here is the problem why filters are not applied on reload. 

settings.filterPokemonBySizeOptions = Store.get('filterPokemonBySizeOptions')

**Comment by JimBim89** [comment]
Also noticed that that size filtering is still active after disabling the pokemon stats toogle.

File: static/js/map/map.settings.js (hunk 1 above)
Comment (Karl-Parris) [reply static/js/map/map.settings.js]: Nice catch. Have made these changes, but can't actually test it at the moment.

**Comment by wraythex** [comment]
Hello, I'm helping out with merging PR's.
This looks like a good PR, thank you for creating it.

I do have some changes to request before I merge this, these are mostly code formatting requests.

Once we have the format matching, would you also be able to create a PR against the async branch?

File: static/js/map/map.pokemon.js [hunk 2]
```diff
@@ -54,6 +54,11 @@ function isPokemonMeetsFilters(pokemon, isNotifPokemon) {
         }
     }
 
+    
+    if(settings.filterPokemonBySize && settings.filterPokemonBySizeOptions && !settings.filterPokemonBySizeOptions.includes(pokemon.size)){
```
Comment (wraythex) [inline static/js/map/map.pokemon.js:58]: On this line, can you put a space between the "if" and the "open parenthesis"?
And also, put a space between the last "close parenthesis" and the "open curly brace"?

File: static/js/map/map.pokemon.js [hunk 3]
```diff
@@ -268,6 +274,17 @@ function pokemonLabel(item) {
                 </div>`
         }
 
+        
+        var sizeDisplay = ''
```
Comment (wraythex) [inline static/js/map/map.pokemon.js:278]: Can you move this variable declaration up to the others?  (next to var statsDisplay = '' and friends)

File: static/js/map/map.pokemon.js [hunk 4]
```diff
@@ -268,6 +274,17 @@ function pokemonLabel(item) {
...
+        if(size){
```
Comment (wraythex) [inline static/js/map/map.pokemon.js:279]: On this line, can you put a space between the "if" and the "open parenthesis"?
And also, put a space between the last "close parenthesis" and the "open curly brace"?

File: static/js/map/map.pokemon.js [hunk 5]
```diff
@@ -630,3 +648,17 @@ function sendPokemonNotification(pokemon) {
     notificationData.cp_multiplier = pokemon.cp_multiplier
     notifiedPokemonData[pokemon.encounter_id] = notificationData
 }
+
+function getSizeDisplay(size){
```
Comment (wraythex) [inline static/js/map/map.pokemon.js:652]: Can you move this function to the utils.pokemon.js file?
Once it is there, please list it in the "exported" comment section at the top of the utils.pokemon.js file.
And, list it in the "globals" comment section at the top of the map.pokemon.js file.

File: static/js/map/map.pokemon.js [hunk 6]
```diff
@@ -630,3 +648,17 @@ function sendPokemonNotification(pokemon) {
...
+       return getSizes()[size] ?? null
+}
+
+function getSizes(){
```
Comment (wraythex) [inline static/js/map/map.pokemon.js:656]: I don't believe this needs to be a function.
Can you create a static array near the top of the utils.pokemon.js file. Something like the following line:
const pokemonSizes = ['Unknown', 'XXS', 'XS', 'M', 'XL', 'XXL']

The "getSizeDisplay" function will need to be updated for this change.

File: static/js/map/map.pokemon.js [hunk 7]
```diff
@@ -630,3 +648,17 @@ function sendPokemonNotification(pokemon) {
     notificationData.cp_multiplier = pokemon.cp_multiplier
     notifiedPokemonData[pokemon.encounter_id] = notificationData
 }
+
+function getSizeDisplay(size){
+       return getSizes()[size] ?? null
```
Comment (wraythex) [inline static/js/map/map.pokemon.js:653]: Can you return an empty string, instead of null here?

File: static/js/map/map.settings.js [hunk 8]
```diff
@@ -231,6 +233,26 @@ function initSettingsSidebar() {
             updatePokemons()
             Store.set('pokemonIconSizeModifier', iconSize)
         })
+
+        $('#filter-by-size-switch').on('change', function () {
+            settings.filterPokemonBySize = this.checked
+            const sizeFilterOptions = $('#filter-by-size-select').closest('.form-control')
```
Comment (wraythex) [inline static/js/map/map.settings.js:239]: Is "closest" needed here?

**Comment by wraythex** [comment]
In addition to the above requested changes, someone on Discord made a good point.

In map.pokemon.js, can you remove the <div> that contains the following line:
${i18n('Weight')}: <strong>${weight}kg</strong> | ${i18n('Height')}: <strong>${height}m</strong>

And also, could you remove the two associated variables "var height" and "var weight".

Since height and weight are random per trainer now, this line no longer makes sense to have on the map.

**Pull request closed without merging**